src/SimpleSAML/Utils/System.php

```diff
@@ -187,7 +187,7 @@ public function resolvePath(string $path, string $base = null): string
      */
     public function writeFile(string $filename, string $data, int $mode = 0600): void
     {
-        $tmpFile = $this->getTempDir() . DIRECTORY_SEPARATOR . rand();
+        $tmpFile = $filename . '.' . bin2hex(openssl_random_pseudo_bytes(4));;
```

Comment:
It's not necessarily wrong, but I wouldn't use an `openssl_` function for non-cryptographic actions. Using `random_bytes(4)` seems more appropriate to use here.